src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[3];
+  const oldDateFormat = {};
+  const separatedDate = date.split(separator);
+
+  fromFormat.slice(0, 3).forEach((key, index) => {
+    oldDateFormat[key] = separatedDate[index];
+  });
+
+  if (fromFormat.includes('YY')) {
+    oldDateFormat['YYYY'] =
+      parseInt(oldDateFormat['YY'], 10) < 30
+        ? '20' + oldDateFormat['YY']
+        : '19' + oldDateFormat['YY'];
```

Comment:
The logic for converting 'YY' to 'YYYY' assumes that any year less than 30 is in the 2000s, and any year 30 or greater is in the 1900s. This assumption may not hold for all use cases, especially if the input date could be from a different century. Consider making this logic more flexible or configurable.